documentation/src/main/asciidoc/topics/con_marshalled_object_encoding.adoc

```diff
@@ -0,0 +1,24 @@
+[id='marshalled-object-encoding_{context}']
+= Marshalled Java objects
+{brandname} stores marshalled Java objects in caches as wrapped `byte[]` arrays.
+For example, the following is a simple representation of a `Person` object stored in memory:
+
+[source,text,options="nowrap",subs=attributes+]
+----
+key=[61 6c 61 6e 0a]
+value={data=ByteArray{array=[70 61 72 74 72 69 64 67 65]}}
+----
+
+To store marshalled objects in caches, you should use the ProtoStream marshaller unless a strict requirement exists.
+For example, when migrating client applications from older versions of {brandname}, you might need to temporarily use JBoss marshalling with your Hot Rod Java clients.
+
+{brandname} stores marshalled Java objects as byte arrays with the following media types:
+
+* `application/x-protostream`
+* `application/x-jboss-marshalling`
+* `application/x-java-serialized-object`
+
+.Equality
+
+It is generally sensible to implement an `equals()` method on your objects and compare equality when storing in binary formats.
+The result of the `equals()` method for a key or value should be consistent in comparison with the serialized key or value.
```

Comment:
Is there a reason for including this section? I think this will just confuse users tbh and I'm not sure what benefit it has. The `.equals()` method has no impact on the how the objects are serialized AFAIK.

Reply:
here's a link to the current form: https://infinispan.org/docs/stable/titles/developing/developing.html#storing_binary_objects-data 

If you look in the "Equality Considerations" block. I've tried to whittle it down based on a comment from @danberindei but maybe it's too low-level. This topic is more about how entries are encoded to caches rather than strictly being about marshalling. I know those two things are closely related but maybe this bit of info makes more sense if you just look at the docs in terms of "encoding" rather than "marshalling".

Reply:
There's some confusion here and I don't think your changes reflect the original intent of the link. The section you linked is discussing the wrapping of `byte[]` as `WrappedBytes` and how comparing multiple `WrappedBytes` instances can result in `!equals` depending on whether the wrapped instance has been serialized or deserialized. Tbh I don't know how to make this section more accessible as it's a very low-level detail and I'm not sure how big an impact it will have  on the majority of users.

Reply:
I'm in favour of dropping this from docs as being too low-level. maybe it would make sense to add the original details to the Javadocs for https://docs.jboss.org/infinispan/12.1/apidocs/org/infinispan/commons/marshall/WrappedBytes.html 

@wburns I believe you are the original author of the "Equality considerations" block. How do you feel about moving that detail into the Javadoc?

Reply:
I am not sure how this section has become like this. In fact the whole second paragraph is not even true, maybe it was from something a long time ago? The only piece of pertinent information for the user here is the last line of the first paragraph in my opinion. That line is `This has an effect on the behavior of equality, which is important to note if you implement an equals() methods on keys.` At most we should mention when using some type of marshalled format that the equality of the object is no longer utilized for testing equals or hash code and the resulting bytes are used instead. 

There is also the fact that instance equality (ie. `==`) should never be relied upon even in local cases as the data may be marshalled, but that is probably a bit out of scope in my opinion.

Reply:
thanks @wburns and @ryanemerson I've reduced it down to a note based on Will's comment. thanks for sharing the knowledge! I'll mark as resolved but do give a shout if you've nits. cheers.